### PR TITLE
Fix/improve layout

### PR DESCRIPTION
**Pull request by Rhian**
This is a fork of https://github.com/Think-and-Dev/solidoc which is a documentation solution for smart contracts. 

The changes in this PR update the module to document contracts up to v0. 8.14 of Solidity. In addition to providing support for documenting structs, it also includes the following updates:

- Better support for Events, which were intermittently missed before
- Enhanced presentation of return values (now tabulated instead of free text)
- Improved presentation of Constructor documentation

To test the output, follow the instructions in this draft PR: https://github.com/bosonprotocol/boson-protocol-contracts/pull/57

Once this PR is approved and merged, I will update the version in the `package.json`, add an explainer line in the `README` and push to the Boson registry.

### CODE REVIEW

**Review by cliffhall — COMMENTED** [review]
* We should add a `.gitignore `with `.DS_Store` in it.

**Comment by zajck** [comment]
I found some layout problems with [structs](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes/#structs)

image

1. First struct field should also be indented.
2. Solidity uses the `dataType variableName` style, not the other way around as it is now. I personally don't care that much, but I think it would be easier for other developers if we used the solidity style.
3. When enum types are part of a struct, they include the word "enum", which again is not how it's done in the solidity (where enum is omitted). I think that that should be removed for consistency.

The example from the picture above would then look like this:
```
struct AuthToken {
    uint256 tokenId;
    BosonTypes.AuthTokenType tokenType;
}
```

**Comment by cliffhall** [comment]
Screen Shot 2022-10-04 at 3 27 19 PM

Another small struct issue, extra space before the commas.

**Review by cliffhall — DISMISSED** [review]
LGTM! 👍

**Comment by Rhian** [comment]
> Just noticed, you will also need to change the package name in package.json. It's currently `@thinkanddev/solidoc`. I'm sure you'll run into it. I suppose `@bosonprotocol/solidoc`?

Fixed

**Comment by Rhian** [comment]
> * We should add a `.gitignore `with `.DS_Store` in it.

Fixed in `0d9deda1891bd97667f37689cdc8a64e3f62378c`

**Comment by Rhian** [comment]
> Screen Shot 2022-10-04 at 3 27 19 PM
> 
> Another small struct issue, extra space before the commas.

Fixed - please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes

**Comment by Rhian** [comment]
> I found some layout problems with [structs](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes/#structs)
> 
> image
> 
> 1. First struct field should also be indented.
> 2. Solidity uses the `dataType variableName` style, not the other way around as it is now. I personally don't care that much, but I think it would be easier for other developers if we used the solidity style.
> 3. When enum types are part of a struct, they include the word "enum", which again is not how it's done in the solidity (where enum is omitted). I think that that should be removed for consistency.
> 
> The example from the picture above would then look like this:
> 
> ```
> struct AuthToken {
>     uint256 tokenId;
>     BosonTypes.AuthTokenType tokenType;
> }
> ```

Fixed - please validate here: Fixed - please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-BosonTypes

File: builders/argument-builder.js [hunk 1]
```diff
@@ -25,11 +25,11 @@ module.exports = {
       builder.push(" | ");
       description = documentationHelper.get(documentation, "param " + parameter.name);
 
-      description.includes(" - ") ?
+      description.includes("-") ?
         description.replace(" - ", "").trim() :
```
[the remark this answers is not in the document]
Comment (Rhian) [reply builders/argument-builder.js]: Fixed.

**Comment by Rhian** [comment]
> I read other comments and I think these two were not reported yet:
> 
> 1. unnamed return areguments are not proper rendered. A lot of examples in `ConfigHandlerFacet`
>    Code:
>    image
>    Documentation: ([Link to the example](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-ConfigHandlerFacet#gettokenaddress))
>    image
> 2. Dashes in the description of parameters are not removed. We use dash between variable name and the description both in return arguments and params. For returns dash is removed, while for param it is not.
>    [Link to the example](https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonGroupHandler#getgroup)
>    image

Fixed - please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonConfigHandler#gettokenaddress

**Comment by Rhian** [comment]
> At the top of the contract files we have a comment with an @title tag and an @notice tag. For some contracts, the @notice text is rendered (e.g. https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-FundsHandlerFacet/ and https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-DisputeHandlerFacet), but for most it is not (e.g. https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-ExchangeHandlerFacet and https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-AgentHandlerFacet)

Fixed in `3c8d0791d0a31d12ae0d5d92848b14025b18207e`

**Comment by Rhian** [comment]
> Adding this comment from slack: I noticed that the description for modifier arguments is not being rendered, and the type between the brackets isn’t linked to the type page like it is for function arguments. See https://docs-website-cadmium.bsn-development-potassium.bosonprotocol.io/docs/contracts/version-undefined-BeaconClientBase/ onlyRole modifier

We no longer have this file on the website, but I've fixed and tested locally, as shown here. 

modifiers

**Comment by Rhian** [comment]
> getAgent code getAgent Documentation I noticed the Description is missing for arguments. I do see the Description for return values.

Please validate here: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonDisputeHandler

**Comment by Rhian** [comment]
Fixed in `bc9414f87e72d876d606c5cef63681f7008e47f0`

**Comment by cliffhall** [comment]
Screen Shot 2022-10-13 at 9 21 10 AM

I see the "space before comma" fix worked, but just noticed that, like enums previously, when included in a struct, another struct has the word 'stuct' brefore the type that should be removed.

**Comment by cliffhall** [comment]
Screen Shot 2022-10-13 at 9 27 39 AM

I see that the "description missing from arguments table" is fixed, but then noticed this one weird instance, the name and descriptions are placeholders or something, but they have actual text in the natspec.

Also descriptions are not shown for return values that are not structs, i.e., normal types
Screen Shot 2022-10-13 at 9 29 42 AM

**Comment by Rhian** [comment]
> Screen Shot 2022-10-13 at 9 27 39 AM
> 
> I see that the "description missing from arguments table" is fixed, but then noticed this one weird instance, the name and descriptions are placeholders or something, but they have actual text in the natspec.
> 
> Also descriptions are not shown for return values that are not structs, i.e., normal types Screen Shot 2022-10-13 at 9 29 42 AM

Please re-review: https://docs.bosonprotocol.io/docs/technical_documentation/contracts/version-undefined-IBosonDisputeHandler#getdispute

**Review by cliffhall — APPROVED** [review]
LGTM! 👍

**Pull request merged**